numpy/_core/src/umath/ufunc_operations.cpp

```diff
@@ -8,54 +8,56 @@
 using namespace np;
 
 template <typename T>
-static void op_ones_like(Slice<T>, Slice<T> &out)
-{
-    for (auto &a : out) {
-        if constexpr (std::is_same_v<T, Half>) {
-            a = Half::FromBits(0x3c00u);
-        }
-        else {
-            a = 1;
+struct OnesLikeOP {
```

Comment:
Bit unclear, at least just following the commits, why this indirection was made. I liked the first iteration better in that one is just writing an inner loop function, which seems more natural. But I guess this where documentation would come in...

Reply:
I'm also curious 😸 a meta object of templated functions doesn't seem functionally different from the struct variant, which probably means I'm missing something obvious.